benchmark/single-source/IndexPathTest.swift

```diff
@@ -0,0 +1,130 @@
+//===--- IndexPathTest.swift ----------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import TestsUtils
+
+let size = 200
+let increasingIndexPath = indexPath(size)
+let decreasingIndexPath = indexPath(size, reversed: true)
+let increasingMaxMiddleIndexPath = indexPath(size, middle: size + 1)
+let increasingMinMiddleIndexPath = indexPath(size, middle: -1)
+let tags: [BenchmarkCategory] = [.validation, .api, .IndexPath]
+
+public let IndexPathTest = [
+  BenchmarkInfo(
+    name: "IndexPath.Subscript.Mutation",
+    runFunction: { n in
+      run_IndexPathSubscriptMutation(n * 10, size, increasingIndexPath)
+    },
+    tags: tags,
+    setUpFunction: { blackHole(increasingIndexPath) }),
+  BenchmarkInfo(
+    name: "IndexPath.Subscript.Range.Mutation",
+    runFunction: { n in
+      run_IndexPathSubscriptRangeMutation(n, size, increasingIndexPath)
+    },
+    tags: tags,
+    setUpFunction: { blackHole(increasingIndexPath) }),
+  BenchmarkInfo(
+    name: "IndexPath.Max",
+    runFunction: { n in run_IndexPathMax(n * 10) },
+    tags: tags),
```

Comment:
It would be good to add a setUpFunction containing "blackHoles" for the used global variables for this benchmark.